### PR TITLE
Fix for 17497

### PR DESCRIPTION
**Pull request by ihenshaw**
Changed runAndroid.js to generate .packager.bat and launchPackager.ba…t to call it to setup the environment variable

## Motivation

Trying to use react-native on a Windows box with a virus killer that runs on port 8081...

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)
1) Start an android emulator
2) on a react-native project (with the changes), run `react-native run-android --port 9988`. 
3) When the packager starts, verify that it states the correct port in the terminal window.
4) verify that the application correctly starts in the Emulator.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/react-native-website, and link to your PR here.)

## Release Notes

[INTERNAL] [BUGFIX] [./scripts] - Fixed runAndroid to enable the use of a package on port <> 8081 for Windows.

### CODE REVIEW

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -263,17 +263,18 @@ function runOnAllDevices(args, cmd, packageNameWithSuffix, packageName, adbPath)
 }
 
 function startServerInNewWindow(port) {
-  const scriptFile = /^win/.test(process.platform) ?
+  const isWin = /^win/.test(process.platform);
```
Comment (shergin) [inline local-cli/runAndroid/runAndroid.js]: Please, don't use abbreviations.

File: local-cli/runAndroid/runAndroid.js [hunk 2]
```diff
@@ -263,17 +263,18 @@ function runOnAllDevices(args, cmd, packageNameWithSuffix, packageName, adbPath)
...
     'launchPackager.command';
   const scriptsDir = path.resolve(__dirname, '..', '..', 'scripts');
   const launchPackagerScript = path.resolve(scriptsDir, scriptFile);
   const procConfig = {cwd: scriptsDir};
   const terminal = process.env.REACT_TERMINAL;
 
-  // setup the .packager.env file to ensure the packager starts on the right port
-  const packagerEnvFile = path.join(__dirname, '..', '..', 'scripts', '.packager.env');
-  const content = `export RCT_METRO_PORT=${port}`;
+  // setup the .packager.(env|bat) file to ensure the packager starts on the right port
```
Comment (shergin) [inline local-cli/runAndroid/runAndroid.js]: Can we decouple file names and command names into separate variables/constants?

File: local-cli/runAndroid/runAndroid.js (hunk 1 above)
Comment (ihenshaw) [reply local-cli/runAndroid/runAndroid.js]: Renamed `isWin` to `isWindows` everywhere where it's necessary.

File: local-cli/runAndroid/runAndroid.js (hunk 2 above)
Comment (ihenshaw) [reply local-cli/runAndroid/runAndroid.js]: I'm not sure what you want here? are you looking for new `const` variables, one for `.packager.bat` and one for `.packager.env`?

I'd have preferred to use just a single filename, unfortunately, Windows doesn't have a `source .env` equivalent, you can get the same result by using a batch file, but the batch files must have a .bat extension to be invoked with call from the parent script...

**Pull request closed without merging**